lib/_007/Val.pm

```diff
@@ -40,9 +40,24 @@ constant TYPE = hash();
 
 class _007::Type {
     has Str $.name;
+    has $.base = TYPE<Object>;
     has @.fields;
     # XXX: $.id
 
+    method install-base($none) {
```

Comment:
I don't think that param should be called `none`

Reply:
I went back and forth on the name of the method (calling it `lift-object-base` in an earlier iteration), but `$none` happens to be a very descriptive name of the parameter. We only need to call this method exactly once. In all other cases we can set the base directly when creating a new type because its base type already exists. The exception is `Object`, whose base type is `None` (yes, a value, not a type). `None` is created a fair bit later in the file, and so we pass it to the method there to set the base of `Object`.